odoo/addons/base/models/ir_qweb_fields.py

```diff
@@ -639,13 +639,26 @@ def value_to_html(self, value, options):
             v, r = divmod(r, secs_per_unit)
             if not v:
                 continue
-            section = babel.dates.format_timedelta(
-                v*secs_per_unit,
-                granularity=round_to,
-                add_direction=options.get('add_direction'),
-                format=options.get('format', 'long'),
-                threshold=1,
-                locale=locale)
+            try:
+                section = babel.dates.format_timedelta(
+                    v*secs_per_unit,
+                    granularity=round_to,
+                    add_direction=options.get('add_direction'),
+                    format=options.get('format', 'long'),
+                    threshold=1,
+                    locale=locale)
+            except KeyError:
+                # in case of wrong implementation of babel, try to fallback on en_US locale.
+                # https://github.com/python-babel/babel/pull/827/files
+                # Some bugs already fixed in 2.10 but ubuntu22 is 2.8
+                localeUS = babel_locale_parse('en_US')
```

Comment:
```suggestion
                if babel.__version__ <= '2.10.3':
                    localeUS = babel_locale_parse('en_US')
```
maybe?

Reply:
As discussed in Live, Babel is full of bugs that are fixed from version to version ...
So until we are in Ubuntu 24 where we know that this bug is fixed, we could keep it in all cases/versions...